app/frontend/utils/formio-submission-traversal.ts

```diff
@@ -0,0 +1,33 @@
+import * as R from "ramda"
+import { isFieldSetKey } from "./formio-helpers"
+
+export const getSinglePreviousSubmissionJson = (submissionData: any, requirementKey: string) => {
+  if (isFieldSetKey(requirementKey)) {
+    let data = {}
+    Object.keys(submissionData.data).forEach((sectionKey) => {
+      const section = submissionData.data[sectionKey]
+      if (!R.isEmpty(data)) return
+
+      const filteredObject = Object.keys(section)
+        .filter((key) => key.startsWith(requirementKey))
+        .reduce((filteredObj, key) => {
+          filteredObj[key] = section[key]
+          return filteredObj
+        }, {})
```

Comment:
not necessary, but can use `https://ramdajs.com/docs/#pickBy` to make this cleaner. Alternatively the filter is not required at all, the same logic can be added directly to the `reduce`.

```
const filteredObject = Object.keys(section)
        .reduce((filteredObj, key) => {
          if(!key.startsWith(requirementKey)) return filteredObj

          filteredObj[key] = section[key]
          return filteredObj
        }, {})
```